src/WebJobs.Extensions.DurableTask/DefaultPlatformInformation.cs

```diff
@@ -37,7 +37,14 @@ public DefaultPlatformInformation(INameResolver nameResolver, ILoggerFactory log
             if (!this.cachedEnviromentVariables.TryGetValue(variableName, out value))
             {
                 value = this.nameResolver.Resolve(variableName);
-                this.cachedEnviromentVariables.Add(variableName, value);
+                try
+                {
+                    this.cachedEnviromentVariables.Add(variableName, value);
+                }
+                catch (ArgumentException)
```

Comment:
The proper way to deal with concurrent access to a dictionary is to use a lock or to use a `ConcurrentDictionary<>`. `ArgumentException` is just one of several problems that can happen. I've seen cases in production where concurrent dictionary access can result in internal infinite loops that eat up 100% CPU, something we can't prevent with try/catch.